### ISSUE TITLE
Wazuh Indexer is not Working in E2E Tests

### ISSUE THREAD

**Issue by rafabailon** [wazuh/wazuh-qa#5004]
## Description

The Wazuh Indexer is not working in E2E Tests. I have launched the pipeline several times and the error is always the same.

- https://ci.wazuh.info/job/Test_e2e_system/236/
- https://ci.wazuh.info/job/Test_e2e_system/237/

<details>
  <summary>Pipeline Error</summary>
  <br/>

```bash
10:46:42  hudson.remoting.ProxyException: groovy.lang.MissingMethodException: No signature of method: hudson.AbortException.toCustomString() is applicable for argument types: () values: []
10:46:42  	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:154)
10:46:42  	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:161)
10:46:42  	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:165)
10:46:42  	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
10:46:42  	at WorkflowScript.getCheckrunData(WorkflowScript:364)
10:46:42  	at WorkflowScript.run(WorkflowScript:448)
10:46:42  	at ___cps.transform___(Native Method)
10:46:42  	at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:86)
10:46:42  	at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:113)
10:46:42  	at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixName(FunctionCallBlock.java:78)
10:46:42  	at jdk.internal.reflect.GeneratedMethodAccessor1464.invoke(Unknown Source)
10:46:42  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
10:46:42  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
10:46:42  	at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
10:46:42  	at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
10:46:42  	at com.cloudbees.groovy.cps.Next.step(Next.java:83)
10:46:42  	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
10:46:42  	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
10:46:42  	at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:129)
10:46:42  	at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:268)
10:46:42  	at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
10:46:42  	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:18)
10:46:42  	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:51)
10:46:42  	at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:185)
10:46:42  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:403)
10:46:42  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$400(CpsThreadGroup.java:97)
10:46:42  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:315)
10:46:42  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:279)
10:46:42  	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:67)
10:46:42  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
10:46:42  	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:139)
10:46:42  	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
10:46:42  	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:68)
10:46:42  	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
10:46:42  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
10:46:42  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
10:46:42  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
10:46:42  	at java.base/java.lang.Thread.run(Thread.java:829)
10:46:42  Finished: FAILURE
```

</details>

<details>
  <summary>Indexer Error</summary>
  <br/>

```bash
10:34:54  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (24 retries left).
10:35:00  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (23 retries left).
10:35:06  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (22 retries left).
10:35:12  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (21 retries left).
10:35:18  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (20 retries left).
10:35:24  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (19 retries left).
10:35:30  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (18 retries left).
10:35:36  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (17 retries left).
10:35:42  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (16 retries left).
10:35:48  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (15 retries left).
10:35:54  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (14 retries left).
10:36:00  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (13 retries left).
10:36:06  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (12 retries left).
10:36:12  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (11 retries left).
10:36:18  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (10 retries left).
10:36:24  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (9 retries left).
10:36:30  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (8 retries left).
10:36:36  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (7 retries left).
10:36:42  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (6 retries left).
10:36:48  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (5 retries left).
10:36:54  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (4 retries left).
10:37:00  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (3 retries left).
10:37:06  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (2 retries left).
10:37:12  FAILED - RETRYING: [wi1]: Wait for Wazuh indexer API (Private IP) (1 retries left).
10:37:18  
10:37:18  TASK [../roles/wazuh/wazuh-indexer : Wait for Wazuh indexer API (Private IP)] ***
10:37:18  [WARNING]: Module did not set no_log for password
10:37:18  fatal: [wi1]: FAILED! => {
10:37:18      "attempts": 24,
10:37:18      "changed": false,
10:37:18      "content": "1708594638 09:37:18 wazuh yellow 1 1 true 5 5 0 0 1 0 - 83.3%\n",
10:37:18      "content_length": "62",
10:37:18      "content_type": "text/plain; charset=UTF-8",
10:37:18      "cookies": {},
10:37:18      "cookies_string": "",
10:37:18      "elapsed": 0,
10:37:18      "redirected": false,
10:37:18      "status": 200,
10:37:18      "url": "https://172.31.10.57:9200/_cat/health/"
10:37:18  }
```

</details>

## Tasks

- [ ] Investigate the Error
- [ ] Fix the Error

**Comment by Rebits** [comment]
This seems to be an erratic behaviour. 
We can see that this task has correctly running in this build https://ci.wazuh.info/job/Test_e2e_system/239/

However it has been detected issues connecting with macOS agent. Related issue here https://github.com/wazuh/wazuh-jenkins/issues/6246

**Comment by Rebits** [comment]
The issue has been resolved in the 4.8.0 beta 2 release. Further investigation into the matter is no longer necessary.

For a demonstration of the correct environment provisioning, please refer to this example: [Wazuh QA Environment Provisioning](https://ci.wazuh.info/job/Wazuh_QA_environment/982/).

I recommend closing this issue. It can be reopened if the problem resurfaces persistently.

**Comment by rauldpm** [comment]
@Rebits the issue does not exist, please fix the link, however, the error is:

```
Deployment error[Test_e2e_system_239_20240222180855_macos_1400_7_0:org.wazuh.JenkinsException: Maximum instance number reached. Timeout. Architecture: arm64v8]
```

It is related to the MacStadium capacity to run VMs, so LGTM, issue will be reopened if necessary
